src/test/java/org/openlmis/integration/dhis2/service/PeriodGeneratorServiceTest.java

```diff
@@ -95,4 +110,37 @@ public void shouldGenerateCorrectMonthlyTimeRange() {
     assertThat(range.getSecond(), is(februaryFirst));
   }
 
+  @Test
+  public void shouldGenerateCorrectRangeFromGivenPeriodMapping() {
+    PeriodMapping periodMapping = new PeriodMapping();
+    UUID processingPeriodId = UUID.fromString("1649efa3-70df-4b60-a27a-f8017b6389a0");
+    periodMapping.setProcessingPeriodId(processingPeriodId);
+
+    ProcessingPeriodDto processingPeriodDto = new ProcessingPeriodDto();
+    processingPeriodDto.setStartDate(new Date(1000));
+    processingPeriodDto.setEndDate(new Date(30000));
+    when(referenceDataService.findProcessingPeriod(processingPeriodId))
+            .thenReturn(processingPeriodDto);
+
+    Pair<ZonedDateTime, ZonedDateTime> range = periodGeneratorService.generateRange(periodMapping);
+
+    ZonedDateTime startDate = ZonedDateTime
+            .ofInstant(processingPeriodDto.getStartDate().toInstant(), ZoneId.systemDefault());
+    ZonedDateTime endDate = ZonedDateTime
+            .ofInstant(processingPeriodDto.getEndDate().toInstant(), ZoneId.systemDefault());
+    assertThat(range, is(Pair.of(startDate, endDate)));
+  }
+
+  @Test
+  public void shouldFormatDateBasedOnGivenDhisPeriodType() {
+    DhisPeriodType dhisPeriodType = new DhisPeriodType();
+    dhisPeriodType.setIsoFormat("yyyyMM");
+    ZonedDateTime zonedDateTime =
+            ZonedDateTime.parse("2011-10-02T14:45:30.123456789+05:30[Asia/Kolkata]");
+
+    String result = periodGeneratorService.formatDate(zonedDateTime, dhisPeriodType);
+
+    assertEquals(result, "201110");
```

Comment:
expected on the left side

Reply:
I swapped the expected and actual arguments.